packages/api/cms-api/src/dependencies/dependencies.service.ts

```diff
@@ -64,17 +71,65 @@ export class DependenciesService {
         console.time("creating block dependency materialized view");
         await this.connection.execute(`DROP MATERIALIZED VIEW IF EXISTS block_index_dependencies`);
         await this.connection.execute(`CREATE MATERIALIZED VIEW block_index_dependencies AS ${viewSql}`);
+        await this.connection.execute(
+            `CREATE UNIQUE INDEX ON block_index_dependencies ("rootId", "rootTableName", "rootColumnName", "blockname", "jsonPath", "targetTableName", "targetId")`,
+        );
         console.timeEnd("creating block dependency materialized view");
 
         console.time("creating block dependency materialized view index");
         await this.connection.execute(`CREATE INDEX block_index_dependencies_targetId ON block_index_dependencies ("targetId")`);
         console.timeEnd("creating block dependency materialized view index");
     }
 
-    async refreshViews(): Promise<void> {
-        console.time("refresh materialized block dependency");
-        await this.connection.execute("REFRESH MATERIALIZED VIEW block_index_dependencies");
-        console.timeEnd("refresh materialized block dependency");
+    async refreshViews(options?: { force?: boolean; consoleCommand?: boolean }): Promise<void> {
+        const refresh = async (options?: { concurrently: boolean }) => {
+            console.time("refresh materialized block dependency");
+            const blockIndexRefresh = this.refreshRepository.create({ startedAt: new Date() });
+            await this.refreshRepository.getEntityManager().persistAndFlush(blockIndexRefresh);
+
+            await this.connection.execute(`REFRESH MATERIALIZED VIEW ${options?.concurrently ? "CONCURRENTLY" : ""} block_index_dependencies`);
+
+            await this.refreshRepository.getEntityManager().persistAndFlush(Object.assign(blockIndexRefresh, { finishedAt: new Date() }));
+            console.timeEnd("refresh materialized block dependency");
+        };
+
+        if (options?.force) {
+            // force refresh -> refresh sync
+            await this.refreshRepository.qb().truncate();
+            await refresh();
+            return;
+        }
+
+        const lastRefreshes = await this.refreshRepository.find({}, { orderBy: { finishedAt: QueryOrder.DESC_NULLS_FIRST }, limit: 1 });
```

Comment:
I think you should additionally order by startedAt? Or is it guaranteed that there is only a single entry with finishedAt=null?

Reply:
I ordered first by finishedAt then by startedAt